fdbserver/TLogServer.actor.cpp

```diff
@@ -2424,6 +2428,32 @@ ACTOR Future<UID> getClusterId(TLogData* self) {
 	}
 }
 
+ACTOR Future<EncryptionAtRestMode> getEncryptionAtRestMode(TLogData* self) {
+	state ReadYourWritesTransaction tr(self->cx);
+	loop {
+		try {
+			tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+			Optional<Value> value = wait(tr.get(encryptionAtRestModeConfKey));
+			EncryptionAtRestMode m;
+			if (value.present()) {
+				m = EncryptionAtRestMode::fromValue(value);
+			}
+
+			// TODO: Encryption at-rest is controlled via SEVER_KNOBs, in future the code shall be updated to replace
+			// per-component (TLog, Redwood, BlobGranule) KNOBs with single db-config
+			if (m.mode == EncryptionAtRestMode::Mode::DISABLED &&
```

Comment:
What if m.mode is enabled but the server knobs have encryption disabled?

Reply:
> What if m.mode is enabled but the server knobs have encryption disabled? Is it fine to leave encryption enabled? 

Good point, db-config can be set as part of `new cluster creation`; the current behavior would `enable` encryption if `db-config` is _not_ set, given KNOBs are the acceptable modes to control the feature, it is acceptable, however, if the cluster is deployed with `encryption enabled` config set, then we don't want to override the config for sure.

Reply:
But what if the db-config has set it to `false` and the knobs want to set it to `true`. Wouldn't we override the config then?

Reply:
> But what if the db-config has set it to Disabled and the knobs want to enable it. Wouldn't we override the config then?

In current code, KNOBs are the one controlling the feature, in some sense, `db config` isn't used to enable/disable encryption. IMO, overriding config using knobs seems like a reasonable thing, till we get around _removing_ knobs and control the feature only via `db-config`. Addition of `db config` was a step forward in that direction, but, we wanted to keep the KNOBs around as it allows better testing capabilities (per-component testing). I would update to `comment db-config read` for now and we can enable it later. I think it would make this code more readable and less confusing. 

Thanks